### DISCUSSION TITLE
CI_DEPENDENCY_PROXY_GROUP_IMAGE_PREFIX seems to not be handler correctly

### DISCUSSION THREAD

**Discussion by andyinno** [renovatebot/renovate#19951] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

34.107

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### If you're self-hosting Renovate, tell us what version of the platform you run.

Gitlab CE

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

From the documentation I can see that CI_DEPENDENCY_PROXY_GROUP_IMAGE_PREFIX should be removed while trying to get the updated information for the docker image that I am using.

As an example I get:

```
Renovate failed to look up the following dependencies: renovate/renovate.

Files affected: .gitlab-ci.yml
```

In the dependency dashboard issue that renovate creates.

My gitlab-ci file relevant part is:

```
image: 
  name: ${CI_DEPENDENCY_PROXY_GROUP_IMAGE_PREFIX}/renovate/renovate:34@sha256:6abf7738395aa40820f93dc5891d5b1c746da3a82669a1e1ffe70f7445e74a4b
  pull_policy: if-not-present
```

If I execute renovate with debug info on, in the log I see:

```
DEBUG: Failed to look up docker dependency renovate/renovate (repository=andrea-work/renovate/renovate-runner, packageFile=.gitlab-ci.yml, dependency=renovate/renovate)

```
and some line below it:

```
"deps": [
               {
                 "depName": "renovate/renovate",
                 "currentValue": "34",
                 "currentDigest": "sha256:6abf7738395aa40820f93dc5891d5b1c746da3a82669a1e1ffe70f7445e74a4b",
                 "replaceString": "${CI_DEPENDENCY_PROXY_GROUP_IMAGE_PREFIX}/renovate/renovate:34@sha256:6abf7738395aa40820f93dc5891d5b1c746da3a82669a1e1ffe70f7445e74a4b",
                 "autoReplaceStringTemplate": "${CI_DEPENDENCY_PROXY_GROUP_IMAGE_PREFIX}/{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
                 "datasource": "docker",
                 "depType": "image-name",
                 "depIndex": 0,
                 "updates": [],
                 "warnings": [
                   {
                     "topic": "renovate/renovate",
                     "message": "Failed to look up docker dependency renovate/renovate"
                   }
                 ],
                 "versioning": "docker"
               }
             ]
```
I found the issue #9958 that states that CI_DEPENDENCY_PROXY_GROUP_IMAGE_PREFIX is added in version 32.31.1 but it seems not functional for me. 

I marked the "I never saw this working" because I started using renovate with version 32 (I do know know minor/patch version) and then switched to version 34 (I do know know the correct version because I added the digest to the gitlab-ci config file as a test in order to see if I could detect changes by using the digest. 

If there is some test that I can do, I am more than happy to test.

thank you
Andrea

### Relevant debug logs

<details><summary>Logs</summary>

```
Sorry, I put them in the previous part of this issue
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by viceice** [comment, marked as answer]
renovate detects that, but it uses `index.docker.io` to check for updates, as it doesn't know the value the the variable.

So your issue is probably a not accessable docker hub.

**Reply by andyinno** [reply]
Hello,

thank you for your reply.

I think that you are referring to 

```
DEBUG: Failed to look up docker dependency renovate/renovate (repository=andrea-work/renovate/renovate-runner, packageFile=.gitlab-ci.yml, dependency=renovate/renovate)
```

I think that docker-hub is accessible, during this exact run I downloaded the renovate version published today:

```
Starting service docker:20.10.22-dind ...
Pulling docker image docker:20.10.22-dind@sha256:5fdc2b7bc40538f569022170d8f0c05e333c5c27db9f378ba972c71206e021ba ...
Using docker image sha256:f2d62af953541f648a51076011ec561df86977f503f1ed1e10ee474566931e19 for docker:20.10.22-dind@sha256:5fdc2b7bc40538f569022170d8f0c05e333c5c27db9f378ba972c71206e021ba with digest docker@sha256:5fdc2b7bc40538f569022170d8f0c05e333c5c27db9f378ba972c71206e021ba ...
```

Per test, I tried to refresh another docker image that is pulled from the docker hub just now in my dependency proxy.

From my understanding looking for digests does not hit the pull rate limit of the HUB. 
How can I check / verify that the index.docker.io is accessible or not?

thank you in advance
Andrea

**Reply by viceice** [reply]
run with debug log level. all requests should be logged at end of each repo

**Reply by andyinno** [reply]
Found it, thank you.

This is related to the exact same run

```
DEBUG: Renovate repository PR statistics (repository=andrea-work/renovate/renovate-runner)
       "stats": {"total": 2, "open": 0, "closed": 0, "merged": 2}
DEBUG: Repository result: done, status: activated, enabled: true, onboarded: true (repository=andrea-work/renovate/renovate-runner)
DEBUG: Repository timing splits (milliseconds) (repository=andrea-work/renovate/renovate-runner)
       "splits": {"init": 3727, "extract": 1181, "lookup": 79, "onboarding": 1, "update": 165},
       "total": 6628
DEBUG: Package cache statistics (repository=andrea-work/renovate/renovate-runner)
       "get": {"count": 1, "avgMs": 4},
       "set": {"count": 0}
DEBUG: http statistics (repository=andrea-work/renovate/renovate-runner)
       "urls": {
         "https://api.github.com/repos/whitesource/merge-confidence/contents/beta.json (GET,200)": 1,
         "https://example.com/api/v4/projects/andrea-work%2Frenovate%2Frenovate-config/repository/branches (GET,200)": 1,
         "https://example.com/api/v4/projects/andrea-work%2Frenovate%2Frenovate-config/repository/files/default.json/raw (GET,200)": 1,
         "https://example.com/api/v4/projects/andrea-work%2Frenovate%2Frenovate-runner/issues (GET,200)": 1,
         "https://example.com/api/v4/projects/andrea-work%2Frenovate%2Frenovate-runner/issues/2 (GET,200)": 2,
         "https://example.com/api/v4/projects/andrea-work%2Frenovate%2Frenovate-runner/issues/2 (PUT,200)": 1,
         "https://example.com/api/v4/projects/andrea-work%2Frenovate%2Frenovate-runner/merge_requests (GET,200)": 1
       },
       "hostStats": {
         "api.github.com": {"requestCount": 1, "requestAvgMs": 352, "queueAvgMs": 0},
         "example.com": {"requestCount": 7, "requestAvgMs": 291, "queueAvgMs": 0}
       },
       "totalRequests": 8
DEBUG: dns cache (repository=andrea-work/renovate/renovate-runner)
       "hosts": []
 INFO: Repository finished (repository=andrea-work/renovate/renovate-runner)
       "cloned": true,
       "durationMs": 6628
```

there's no mention of docker.io in there. If I check other repository that I did not have migrated to the dependency proxy yet, I see

```
     "hostStats": {
         "api.github.com": {"requestCount": 2, "requestAvgMs": 272, "queueAvgMs": 0},
         "auth.docker.io": {"requestCount": 3, "requestAvgMs": 360, "queueAvgMs": 0},
         "index.docker.io": {"requestCount": 3, "requestAvgMs": 391, "queueAvgMs": 1},
         "registry.npmjs.org": {
           "requestCount": 4,
           "requestAvgMs": 905,
           "queueAvgMs": 0
         },
```

therefore in the log, is not there. If it's not there is because it has failed? or because it is not executed?

**Reply by viceice** [reply]
you probably need to check all log lines from that repo. there should be a message with the failure, maybe logged as debug

**Reply by andyinno** [reply]
Hello,

sorry for the late reply but I was unable to access the logs during the weekend.

I checked the logs again, the only rows that are someway related to renovate-runner docker image is:

```
DEBUG: Failed to look up docker dependency renovate/renovate (repository=andrea-work/renovate/renovate-runner, packageFile=.gitlab-ci.yml, dependency=renovate/renovate)

```

That does not says anything about the failure reason.

**Reply by andyinno** [reply]
I think I found the issue while reading #19943 that looked somehow similar to my case.

Updating the config.js hostRules allowed me to login correctly to the docker hub. Thing that I was sure I was doing but... no.
As soon as I updated the hostRule the output in the log contains all the docker hub related strings.

Thank you for your support!
